lib/modules/manager/gomod/artifacts.ts

```diff
@@ -191,7 +191,7 @@ export async function updateArtifacts({
     }
   }
   const goConstraints =
-    config.constraints?.go ?? (await getGoConstraints(goModFileName));
+    config.constraints?.go ?? getGoConstraints(massagedGoMod);
```

Comment:
let's do this refactor in a separate PR

Reply:
I'm confused, do you want initial commit? ad5d318

Reply:
What does this mean?
![Screenshot_2025-03-02-04-33-24-035_com brave browser](https://github.com/user-attachments/assets/04ea229a-cdc9-4565-8e28-00354b8d02af)